web/menu.jspf

```diff
@@ -117,7 +121,11 @@ org.opensolaris.opengrok.web.Util"
         if (ph.getGroups().size() > 0 && ph.getAllUngrouped().size() > 0) {
             %><optgroup label="Other"><%
         }
-        for ( Project p : ph.getAllUngrouped()) {
+        for (Project p : ph.getAllUngrouped()) {
```

Comment:
Yes this is exactly the place where the change should be made if necessary. But as you said if empty groups look fine to you just keep it.

For me it would be quite confusing why some groups arent displayed (unauthorized content - empty groups; no projects in the group) and some are (empty group but contains not indexed projects).